dbms/src/Encryption/FileProvider.cpp

```diff
@@ -32,7 +35,15 @@ RandomAccessFilePtr FileProvider::newRandomAccessFile(
     const ReadLimiterPtr & read_limiter,
     int flags) const
 {
-    RandomAccessFilePtr file = std::make_shared<PosixRandomAccessFile>(file_path_, flags, read_limiter);
+    RandomAccessFilePtr file;
+    if (S3::S3FilenameView::fromKey(file_path_).isValid())
```

Comment:
`S3::S3FilenameView::fromKey(file_path_).isValid()` seems too dangerous for me, it could unexpectedly parse a local path as S3 key.

What about adding a prefix "s3://" for the key in S3, and adding these methods for generating/parsing?

*`S3::S3Filename::toFullKeyWithPrefix(file_path_)`
*`S3::S3FilenameView::fromKeyWithPrefix(file_path_)`